plugins/module_utils/net_map/networking_definition.py

```diff
@@ -1654,13 +1765,27 @@ def __parse_raw_net(
             else None
         )
 
+        is_trunk_parent = (
+            bool(templated_net_data[self.__FIELD_IS_TRUNK_PARENT])
+            if self.__FIELD_IS_TRUNK_PARENT in templated_net_data
+            else None
+        )
+
+        trunk_parent = (
+            str(templated_net_data[self.__FIELD_TRUNK_PARENT])
+            if self.__FIELD_TRUNK_PARENT in templated_net_data
+            else None
+        )
+
```

Comment:
I'd suggest doing something similar to what's done here: https://github.com/openstack-k8s-operators/ci-framework/pull/1218/files#diff-35c36b377323c97ce9082f04b63f3dd52f3f32c572cc593a384a393659f1f680R1083
Anyway, may make sense to check if the `trunk_parent` exists, isn't it?

Reply:
I've added validation to check that `trunk_parent` exists. The validation also adds a boundary, the trunk parent port and sub port must be in the same group template.

Reply:
Sounds good.